www/api/.htaccess

```diff
@@ -1,3 +1,6 @@
+Require local
+Require valid-user
+
```

Comment:
This completely breaks normal usage of FPP where a password is not configured.

Reply:
Ah I see. Let me rethink this then. 

Would you accept a PR if I switch this feature on using the SetupHtaccess function?

Reply:
It cannot modify any file in /opt/fpp as it would get wiped out on an upgrade. 

Also note that there are a bunch of calls to fppjson and fppxml from the parts that have to work via localhost.  For example, the OLED based menus on the various controllers currently make calls via fppjson and fppxml.  Until those are changed over to API calls, I don't think we can restrict localhost.

Reply:
Hi Dan, I've pushed a new revision of the changes.

I've modified the .htaccess in www/api to add auth depending on the system configuration with no password / password & external / internal access. 

common/settings.php has been added to. I've specifically called out allow any access to fppjson & fppxml as you said they will be deprecated as & when to move to /api. 

I have tested this in a docker environment & applied the changes on top of 4.6, burned a sd card & tried them on a Raspberry Pi & my tests work as I would expect.

Hopefully this revision gets the green light. Thanks.